lib/AST/ASTMangler.cpp

```diff
@@ -1137,7 +1137,9 @@ void ASTMangler::appendType(Type type, const ValueDecl *forDecl) {
       // type ::= archetype
     case TypeKind::PrimaryArchetype:
     case TypeKind::OpenedArchetype:
-      llvm_unreachable("Cannot mangle free-standing archetypes");
+      // TODO: don't use a dummy type here.
```

Comment:
I'm not sure if/how we should mangle archetypes. Thoughts? Maybe we should try to substitute these?

Reply:
Let's leave the llvm_unreachable in place here. Instead, the caller should map the type that contains archetypes out of context by calling `type->mapTypeOutOfContext()` and mangling that.

Reply:
OK, great. I'll do that.